src/modules/entity/entity.js

```diff
@@ -227,13 +227,20 @@ function drupalgap_entity_render_field(entity_type, entity, field_name,
       // to the entity's content.
       var fn = window[function_name];
       var items = null;
+      // Check to see if translated content based on app's language setting
+      // is present or not. If yes, then use that language as per setting.
       // Determine the language code. Note, multi lingual sites may have a
       // language code on the entity, but still have 'und' on the field, so
       // fall back to 'und' if the field's language code doesn't match the
       // entity's language code.
+
+      var default_lang = language_default();
       var language = entity.language;
       if (entity[field_name]) {
-        if (entity[field_name][language]) {
+        if (entity[field_name][default_lang]) {
+          items = entity[field_name][default_lang];
+        }
+        else if (entity[field_name][language]) {
```

Comment:
@vjsamuel Thanks for another contribution! This does make sense, however I am not a multi lingual site expert... but it seems to me we would want the order of the booleans to be this instead:
```
if (entity[field_name][language]) {
  ...
}
else if (entity[field_name][default_lang]) {
  ...
}
...
```
I say this because it seems the field's language should first try to grab the language from the entity itself, then fallback to the default language, and then finally fall back to `und`. What do you think?